### PR TITLE
feat(console): roles page

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

feat(console): roles page

### Todo
- Pagination
- Search roles
- Create role

## Testing

image

### CODE REVIEW

File: packages/console/src/components/AppContent/components/Sidebar/icons/Role.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import type { SVGProps } from 'react';
+
+const Role = (props: SVGProps<SVGSVGElement>) => {
```
Comment (simeng-li) [inline packages/console/src/components/AppContent/components/Sidebar/icons/Role.tsx:3]: @charIeszhao  are we still manually writing tsx svg component in AC?

Comment (xiaoyijun) [reply packages/console/src/components/AppContent/components/Sidebar/icons/Role.tsx:3]: The existing sidebar icons are all tsx svg components.
I prefer to align with the old code and refactor them together later.

Comment (charIeszhao) [reply packages/console/src/components/AppContent/components/Sidebar/icons/Role.tsx:3]: No, we can directly import an SVG file as a component, with the help of the parcel SVG transformer plugin.

Comment (charIeszhao) [reply packages/console/src/components/AppContent/components/Sidebar/icons/Role.tsx:3]: I don't recommend aligning with the old code as the old one are going to be refactored.

Comment (charIeszhao) [reply packages/console/src/components/AppContent/components/Sidebar/icons/Role.tsx:3]: These svg components are something slipped through the net... which should've been eliminated already

File: packages/console/src/pages/Roles/index.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            title: t('roles.role_name'),
+            dataIndex: 'name',
+            colSpan: 6,
+            render: ({ id, name }) => <ItemPreview title={name} to={buildDetailsPathname(id)} />,
+          },
+          {
+            title: t('roles.role_description'),
+            dataIndex: 'description',
+            colSpan: 10,
+            render: ({ description }) => description,
```
Comment (charIeszhao) [inline packages/console/src/pages/Roles/index.tsx:55]: I think render as simple as this should've been avoided, as we also have dataIndex prop to tell the table component which column data it should use

File: packages/console/src/pages/Roles/index.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+          type="primary"
+          size="large"
+          onClick={() => {
+            navigate(createRolePathname);
+          }}
+        />
+      </div>
+      <Table
+        className={pageStyles.table}
+        rowGroups={[{ key: 'roles', data: roles }]}
```
Comment (charIeszhao) [inline packages/console/src/pages/Roles/index.tsx:40]: nit: Tables that have only 1 row group should avoid requiring the row group key prop. Please help check if we have a linear task to refactor this

**Review by charIeszhao — APPROVED** [review]

File: packages/console/src/pages/Roles/index.tsx (hunk 3 above)
Comment (xiaoyijun) [reply packages/console/src/pages/Roles/index.tsx:40]: LOG-5078

File: packages/console/src/pages/Roles/index.tsx (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/pages/Roles/index.tsx:55]: This will be a big task.
Add an issue to track it: LOG-5079

**Pull request merged**